code/obj/item/device/radio.dm

```diff
@@ -333,29 +333,31 @@ Green Wire: <a href='?src=\ref[src];wires=[WIRE_TRANSMIT]'>[src.wires & WIRE_TRA
 			R.hear_radio(M, messages, lang_id)
 
 	var/list/heard_flock = list() // heard by flockdrones/flockmind
-
+	var/datum/game_mode/conspiracy/N = ticker.mode
+	var/protected_frequency = null
+	if(istype(N))
+		protected_frequency = N.agent_radiofreq //groups conspirator frequency as a traitor one and protects it along with nukies
 	// Don't let them monitor Syndie headsets. You can get the radio_brain bioeffect at the start of the round, basically.
-	if (src.protected_radio != 1 && isnull(src.traitorradio))
+	if (src.protected_radio != 1 && isnull(src.traitorradio) && protected_frequency != display_freq )
 		for (var/mob/living/L in radio_brains)
 			receive += L
 
 		for(var/mob/zoldorf/z in the_zoldorf)
 			if(z.client)
 				receive += z
 
-
-		// hi it's me cirr here to shoehorn in another thing
-		// flockdrones and flockmind should hear all channels, but with terrible corruption
-		if(length(flocks))
-			for(var/F in flocks)
-				var/datum/flock/flock = flocks[F]
-				if(flock)
-					if(flock.flockmind)
-						heard_flock |= flock.flockmind
-					if(flock.units && flock.units.len > 0)
-						for(var/mob/living/D in flock.units)
-							if(D)
-								heard_flock |= D
+	// hi it's me cirr here to shoehorn in another thing
+	// flockdrones and flockmind should hear all channels, but with terrible corruption
+	if(length(flocks))
```

Comment:
changed behaivor

Reply:
Clarification: Changed behaviour because in original code the flock code was indented to be inside the protected radio check and now it isnot.